### PR TITLE
Ensuring right credentials for kubectl calls.

### LINKED ISSUE #424: Need to ensure that gcloud container does not use the default credentials.
By setting the env variable  `CLOUDSDK_CONTAINER_USE_APPLICATION_DEFAULT_CREDENTIALS` to `false` we can ensure that the container will use gcloud's credentials. We want this behavior because the VS extension provides the credentials directly to gcloud to control what environment is running in.

### PR DESCRIPTION
**Pull request by ivannaranjo**
There's an issue in which `kubectl` will use the application default credentials by default to obtain the credentials for calls into the cluster. If a user has more than one account registered with gcloud the default credentials will be one of the one registered with gcloud. This is also independent of the accounts registered with Visual Studio.

This could be workaround by setting `CLOUDSDK_CONTAINER_USE_APPLICATION_DEFAULT_CREDENTIALS` to `false` except that exposes a bug in how `kubectl` quotes the path to gcloud to invoke it to fetch credentials.

In order to workaround these two issues this PR changes the extension to set `CLOUDSDK_CONTAINER_USE_APPLICATION_DEFAULT_CREDENTIALS` to `true` to force `kubectl` to use the application default credentials and then the extension will also set the value of the `GOOGLE_APPLICATION_CREDENTIALS` environment variable to point to the credentials file that corresponds to the user selected in the extension. This way if `kubectl` will always use the same credentials as the extension no matter what.

This also as it turns out is the right behavior for wrapping `kubectl` and ensuring credentials, as we don't want `kubectl` to use `gcloud` to fetch credentials, we want to supply the credentials to it.

Fixes #424

### CODE REVIEW

File: GoogleCloudExtension/GoogleCloudExtension.GCloud/KubectlWrapper.cs [hunk 1]
```diff
@@ -175,5 +182,20 @@ private static string FormatCommand(string command, KubectlContext context, bool
 
             return $"{command} --kubeconfig=\"{context.ConfigPath}\" {format}";
         }
+
+        /// <summary>
+        /// Returns the environment variables to use to invoke kubectl safely. This environemnt is necessary
+        /// to ensure that the right credentials are used should the access token need to be refreshed.
+        /// </summary>
+        /// <param name="context"></param>
```
Comment (iantalarico) [inline GoogleCloudExtension/GoogleCloudExtension.GCloud/KubectlWrapper.cs]: nit: empty params and returns

File: GoogleCloudExtension/GoogleCloudExtension.GCloud/CommonEnvironmentVariables.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace GoogleCloudExtension.GCloud
+{
+    /// <summary>
+    /// This class contains defintions for the environment variables used the various wrappers.
```
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.GCloud/CommonEnvironmentVariables.cs]: used by ?

File: GoogleCloudExtension/GoogleCloudExtension.GCloud/CommonEnvironmentVariables.cs [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    /// </summary>
+    internal static class CommonEnvironmentVariables
+    {
+        // This variables is used to force gcloud to use application default credentials when generating
+        // the cluster information for the cluster.
+        public const string GCloudContainerUseApplicationDefaultCredentialsVariable = "CLOUDSDK_CONTAINER_USE_APPLICATION_DEFAULT_CREDENTIALS";
```
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.GCloud/CommonEnvironmentVariables.cs]: nit:  Use a shorter name? When it's too long, it's not convenient to use it in other places.

File: GoogleCloudExtension/GoogleCloudExtension.GCloud/KubectlContext.cs [hunk 4]
```diff
@@ -28,9 +29,15 @@ public class KubectlContext : IDisposable
         /// </summary>
         public string ConfigPath { get; private set; }
 
-        public KubectlContext(string configPath)
+        /// <summary>
+        /// Path to the application credentials to use while calling into kubectl.
+        /// </summary>
+        public string CredentialsPath { get; private set; }
+
+        public KubectlContext(string configPath, string credentialsPath)
```
Comment (Deren-Liao) [inline GoogleCloudExtension/GoogleCloudExtension.GCloud/KubectlContext.cs]: Add comment.

File: GoogleCloudExtension/GoogleCloudExtension.GCloud/KubectlWrapper.cs (hunk 1 above)
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension.GCloud/KubectlWrapper.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension.GCloud/CommonEnvironmentVariables.cs (hunk 2 above)
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension.GCloud/CommonEnvironmentVariables.cs]: Done.

File: GoogleCloudExtension/GoogleCloudExtension.GCloud/CommonEnvironmentVariables.cs (hunk 3 above)
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension.GCloud/CommonEnvironmentVariables.cs]: Well, that is the unfortunate name of the environment variable, shortening will not really make the code much more readable.

File: GoogleCloudExtension/GoogleCloudExtension.GCloud/KubectlContext.cs (hunk 4 above)
Comment (ivannaranjo) [inline GoogleCloudExtension/GoogleCloudExtension.GCloud/KubectlContext.cs]: It is a constructor, there's really nothing to comment :).

**Comment by ivannaranjo** [comment]
Thank you for the review. Setting the `GOOGLE_APPLICATION_CREDENTIALS` will only happen during the calls to `kubectl` or the `gcloud container cluster` tools. We actually do need to set the env var because the extension needs to always be in control of what credentials are being used.

It turns out it was good that `kubectl` uses the default credentials because this gives us the chance to override them with the ones for the current user in the extension. If `kubectl` uses the `gcloud` helper then it would be probably be getting the wrong credentials to use.

**Pull request merged**